src/app/pages/adoption/adoption.js

```diff
@@ -75,49 +58,30 @@ function FacultyForm({selectedRole, onPageChange}) {
         return true;
     }
 
-    useEffect(() => {
-        if (currentBook) {
-            formRef.current.submit();
-        }
-    }, [currentBook]);
-
-
-    function onSubmit(form) {
-        const selectedBooks = selectedBooksRef.current;
-        const submitQueue = selectedBooks.slice();
-        const iframe = document.getElementById(form.target);
-
-        // eslint-disable-next-line no-use-before-define
-        const submitAfterDelay = () => window.setTimeout(submitNextBook, 3000);
-
-        function submitNextBook() {
-            const preselectedTitle = firstSearchArgument();
-
-            if (submitQueue.length > 0) {
-                setCurrentBook(submitQueue.shift());
-            } else if (iframe) {
-                setCurrentBook(null);
-                iframe.removeEventListener('load', submitAfterDelay);
-                afterFormSubmit(history, preselectedTitle, selectedBooks);
-            }
-        }
-
-        formRef.current = form;
-        if (iframe) {
-            iframe.addEventListener('load', submitAfterDelay);
-        }
-        submitNextBook();
+    function doSubmit(form) {
+        form.submit();
+        onSubmit();
     }
 
     return (
-        <MultiPageForm
-            validatePage={validatePage} action={webtoleadUrl}
-            onPageChange={onPageChange} onSubmit={onSubmit} debug={debug}
-            submitting={currentBook}
-        >
-            <ContactInfoPage selectedRole={selectedRole} validatorRef={contactValidatorRef} />
-            <BookSelectorPage selectedBooksRef={selectedBooksRef} />
-        </MultiPageForm>
+        <React.Fragment>
+            <FormTarget submitting={submitting} />
+            <MultiPageForm
+                validatePage={validatePage} action={adoptionUrl}
+                onPageChange={onPageChange} onSubmit={doSubmit}
+                submitting={submitting} target="form-target"
+            >
+                <React.Fragment>
+                    <input type="hidden" name="application_source" value="OS Web" />
+                    <input type="hidden" name="position" value={position} />
+                    <input type="hidden" name="role" value="Instructor" />
+                    <input type="hidden" name="lead_source" value="Adoption Form" />
+                    <input type="hidden" name="process_adoptions" value={true} />
+                    <ContactInfo validatorRef={contactValidatorRef} />
+                </React.Fragment>
+                <BookSelectorPage selectedBooksRef={selectedBooksRef} />
+            </MultiPageForm>
```

Comment:
would be nice to extend `MarkupChildren` to handle multiple children, that way you wouldn't need to use React.Fragment everywhere with `MultiPageForm`